tutorials/getting-started/custom-widgets.md

```diff
@@ -0,0 +1,246 @@
+---
+title: "Custom widgets"
+---
+
+## Building custom widgets
+
+You've seen a lot of the widgets that come "in the box" with Apostrophe. But you can do much more by creating your own.
+
+### Custom navigation
+
+A common case: you want to build your own navigation menu. [Apostrophe's page tree is awesome](building-navigation.html) but sometimes you want to "cherrypick" pages from all over the tree, perhaps for a special footer.
+
+You could use a rich text widget and just tell users to add links manually. But they'll just break each time a page is moved around the site. And it's easier for users to mess up the formatting that way. We want something more consistent.
+
+Let's look at some custom widgets that help provide navigation. We'll start with a simple widget that adds a link in a well-formatted way.
+
+#### `link`: the simplest widget
+
+First we'll need a folder for the module:
+
+```bash
```

Comment:
Well, so far we've told people what to type at the prompt. We don't have to
do that, but it offers a certain clarity; my eye is drawn to commands, it's
not drawn to "blah blah make this directory blah blah".

The same people whose skills we can assume if we don't tell them how will
presumably also know how to change it for Windows, no?

This isn't a hill I'd die on, but I'm mildly in favor of showing the
command because it feels more "gritty" to me than talking about creating a
folder.

On Tue, Jul 5, 2016 at 10:01 AM, Jimmy Harrison <notifications@github.com>
wrote:

> In tutorials/getting-started/custom-widgets.md
> <https://github.com/punkave/apostrophe-documentation/pull/15#discussion_r69567621>
> :
>
> > +
> > +You've seen a lot of the widgets that come "in the box" with Apostrophe. But you can do much more by creating your own.
> > +
> > +### Custom navigation
> > +
> > +A common case: you want to build your own navigation menu. [Apostrophe's page tree is awesome](building-navigation.html) but sometimes you want to "cherrypick" pages from all over the tree, perhaps for a special footer.
> > +
> > +You could use a rich text widget and just tell users to add links manually. But they'll just break each time a page is moved around the site. And it's easier for users to mess up the formatting that way. We want something more consistent.
> > +
> > +Let's look at some custom widgets that help provide navigation. We'll start with a simple widget that adds a link in a well-formatted way.
> > +
> > +#### `link`: the simplest widget
> > +
> > +First we'll need a folder for the module:
> > +
> > +```bash
>
> Do we want to be using shell commands that might be OS-specific in our
> tutorial? I think making a folder can be an assumed skill, and this command
> would not work as expected in Windows command prompt (-p isn't a valid
> option, reversing the slashes). I think most of the commands we run will
> likely be OS-agnostic (i.e. anything node), so should we try to keep the
> entire tutorial that way?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/punkave/apostrophe-documentation/pull/15/files/758d508e28dd26bb77ad4869ad70c4cdd051325c#r69567621>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAB9fVL5uWLlnRQobvd31IqCpKxbCp8Xks5qSmPEgaJpZM4I7_d4>
> .
>



-- 


*THOMAS BOUTELL, *SUPPORT LEAD
P'UNK AVENUE | (215) 755-1330  |  punkave.com